src/Sarif.UnitTests/Converters/AndroidStudioConverterTests.cs

```diff
@@ -345,8 +363,14 @@ public void AndroidStudioConverter_ConvertSarifResult_GeneratesLocationWithOnlyP
 
             LocationInfo locationInfo = GetLocationInfoForBuilder(builder);
 
-            locationInfo.Location.Should().Be(expectedLocation);
-            locationInfo.LogicalLocationComponents.SequenceEqual(expectedLogicalLocationComponents).Should().BeTrue();
+            LocationEqualityComparer.Instance.Equals(locationInfo.Location, expectedLocation)
+                .Should().BeTrue();
+
+            locationInfo.LogicalLocationComponents
+                .SequenceEqual(
+                    expectedLogicalLocationComponents,
+                    LogicalLocationComponentEqualityComparer.Instance)
```

Comment:
I like it. I didn't even realize that IntelliSense has gotten smarter about that. It does read better.

So, taken together, we'd have two value-comparison members per class and no extra public types.

``` C#
public class LogicalLocationComponent {
        ...
     public static IEqualityComparer<LogicalLocationComponent> ValueComparer { get; }
     public bool ValueEquals(LogicalLocationComponent other) => ValueComparer.Equals(this, other);
}

That looks pretty nice to me. :)